### ISSUE TITLE
Remove "Box" from menu > Configure map > GPX files... > ADD MORE... > menu >

### ISSUE THREAD

**Issue by p-p-s** [osmandapp/Osmand#7684]
Hi OsmAnd team,
First: thank you for your gorgeous app!

I was tired of juggling with .gpx files between PC and Android mobile. Instead of emailing tracks to myself as many recommend or using MyPhoneExplorer to copy files to the OsmAnd Tracks folder, I thought I rather sync them through box.com as I saw "Box" being offered at    
Hamburger menu > Configure map > GPX files... > ADD MORE... > Hamburger menu >
where I can choose from: Downloads, my device's name for its file system, Box, Bug reports, Drive. 

When I choose Box, navigate to the Box folder, pick a single gpx file (picking multiple is not possible) > OPEN
I get
"OsmAnd cannot import the file. Please check if OsmAnd has permission to read file from its location."

I feel OsmAnd has the folder permissions to its own folders, but probably not to the folders of the box.com app. Could I change that? Help/documentation/how-to videos on YouTube are not very talkative here.
I could even grant super user permission to the OsmAnd application user, but the OsmAnd code would need to make actively use of su and my guess is it does not, as I don't find an OsmAnd request to grant su documented.

My Box installation on Android is for individual Box account holder, not for an enterprise user, i.e. there is no Enterprise Mobility Management (EMM) controlling my device and at https://app.box.com/account I have no access to the "Admin Console" where I could grant a permission for OsmAnd (see: "Content can be opened in external applications: Enables you to control people's ability to open content from Box in other applications using the same operating system" at https://community.box.com/t5/How-To-Guides-for-Mobile/Understanding-Mobile-Security-Settings/ta-p/267).

Most OsmAnd users who also use Box probably use the free-of-charge base plan for individual Box account holder and I guess they all bump into the same issue. The few who use an account under an enterprise plan in most cases probably have their Box being managed by their employer who pays for the plan. The employer's admins will grant access to Microsoft Office apps und other apps they need, but likely not for OsmAnd. So, if I am not mistaken, for almost everybody Box cannot be used with OsmAnd. If true, you should rather remove Box from the offered gpx file sources.

**Comment by vshcherb** [comment]
The things are changing in Android World, the right sequence of steps is to access files from Box-application not from OsmAnd. OsmAnd doesn't have permission to scan other folders and won't have them in future at all, so open Box-app, open GPX within it and select OsmAnd app as an application to open with.

**Comment by p-p-s** [comment]
@vshcherb : Right, your response supports my request to remove "Box" from menu > Configure map > GPX files... > ADD MORE... > menu.
The "Box" link in that menu suggests that it would still be possible to initiate using a GPX file in OsmAnd from OsmAnd, which is not anymore the case as you confirm.

**Comment by dmpr0** [comment]
Hello, @p-p-s!
Could you please provide a screenshot of the screen, for me now it's not clear what you mean.
"Add more" button in Configure map -> GPX dialog must open system default file manager.

**Comment by p-p-s** [comment]
Hi dmpr0,
Have a look at the screenshots[https://app.box.com/s/pkgck9xvmkleqz68kcu5podff0lgho4i].
Hamburger menu > Configure map > GPX files... > 
leads me to the first screenshot. When I then tap
ADD MORE... > and I earlier chose already a file in Box, the same folder in Box is chosen again (which is a good concept). This Box folder is shown in the second screenshot (but this is not the Box application process. It is still the OsmAnd process displaying the folder).
When I there tap a file (or go via that Hamburger menu > to navigate to another Box folder or whatever) I am thrown back to the "Hamburger menu > Configure map" folder darkend and showing the error message (third screenshot).

**Comment by dmpr0** [comment]
Hello, @p-p-s! Thank you for the screenshot.
Please try to clear default apps:
https://support.google.com/android/answer/9415055?hl=en
https://c.mi.com/thread-1052367-1-0.html

**Comment by p-p-s** [comment]
Hi dmpr0,
I added 4 more to the [screenshots](https://app.box.com/s/pkgck9xvmkleqz68kcu5podff0lgho4i).
As you can see, neither Box nor OsmAnd have any defaults set. Resetting the defaults of Box and OsmAnd is not possible. I am not keen on blindly exercising a global defaults reset affecting all apps. 
-Peter
